workspaces/diff-engine/src/events/endpoint.rs

```diff
@@ -1,13 +1,13 @@
 use super::EventContext;
 use cqrs_core::Event;
-use serde::Deserialize;
+use serde::{Deserialize, Serialize};
 
 use crate::state::endpoint::{
   PathComponentId, RequestId, RequestParameterId, ResponseId, ShapedBodyDescriptor,
   ShapedRequestParameterShapeDescriptor,
 };
 
-#[derive(Deserialize, Debug)]
+#[derive(Deserialize, Debug, PartialEq, Serialize)]
 pub enum EndpointEvent {
```

Comment:
All `SpecEvent` variants and their underlying structs became `PartialEq` (so we can express `specA == specB`, so far mostly useful in testing) and `Serialize` (so we can write the events to stdout).